documentation/correctness.adoc

```diff
@@ -0,0 +1,56 @@
+= Model correctness
+
+We can identify a number of levels of correctness of a model.
+Each level builds on the previous one, so achieving a certain level of correctness implies having achieved all previous levels.
+The levels are divided into two layers.
+
+.Model correctness explained in a diagram
+image::../resources/correctness-diagrams/levels.svg[]
+
+
+== Correctness of the JSON serialization of a model
+
+A JSON serialization of a model is considered to be deserializable (precisely) if it's correct with respect to all these levels.
+
+[horizontal]
+Low-level syntactic:: The serialization is parsable as JSON.
+Note that “parsable” doesn't necessarily mean “syntax-correct”.
+E.g., trailing commas are not syntax-correct, but don't necessarily prevent the JSON from being parsable, nor deserializable.
+
+Structural syntactic:: The JSON conforms to (the relevant version of) https://github.com/LIonWeb-org/lioncore-typescript/blob/main/schemas/generic-serialization.schema.json[the general JSON Schema for the serialization format].
+The relevant version should be read off from the value of the `serializationFormatVersion` property of the root object.
+Note that JSON Schema isn't powerful enough to express more high-level, nor referential constraints on the model serialized as JSON.
+
+Hierarchical:: The JSON doesn't violate the following constraints.
++
+* No double containments, i.e. nodes are not the children of multiple other nodes, nor of themselves.
+* The `parent` property of a serialized node matches the `id` of the containing node (whenever that node is present).
+* No (incompatible) double serializations.
+* Node IDs are unique (within the serialization).
++
+Note that links don't have to resolve, so no additional constraints are imposed.
+
+Meta-referential:: The metamodels specified in the serialization are known and _available_ in the sense that the client('s deserializer) knows how to deserialize models conforming to them.
+That might include aspects like type definitions, and metamodel-specific APIs.
```

Comment:
I do not think this is a validation level,  this is a prerequisite to be able to do validations that need the used language defintions.

Reply:
Agreed. Will rewrite a bit.